### PR TITLE
Bazel: Support bazel build for essential modules

### PR DESCRIPTION
**Pull request by golovasteek**
## Describe your changes
In this MR we introduce Bazel build annotations, for the essential.
There is a `everest_cc_module` rule introduced that is used for the module declaration.

The aim of this PR is to create basic infrastructure for the module builds.

The bazel support is still experimental and maintained at best-effort basis.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements 

## TODO before merge:
- [x] Merge https://github.com/EVerest/everest-framework/pull/166
- [x] Merge https://github.com/EVerest/everest-utils/pull/136
- [x] Update references in dependencies.yaml

### CODE REVIEW

**Comment by golovasteek** [comment]
This PR should go only after this one https://github.com/EVerest/everest-framework/pull/162

Before merging we should switch the everest-framework reference to new revision in everest/core/main

File: dependencies.yaml [hunk 1]
```diff
@@ -76,8 +76,8 @@ ext-mbedtls:
     - MBEDTLS_FATAL_WARNINGS OFF  # disables setting warnings as errors  FIXME: workaround until upstream-fixes are included
 # everest-testing
 everest-utils:
-  git: https://github.com/EVerest/everest-utils.git
-  git_tag: v0.2.1
+  git: https://github.com/Qwello/everest-utils.git
```
Comment (hikinggrass) [inline dependencies.yaml:79]: (we should not forget to switch this back to EVerest)

**Comment by SirVer** [comment]
This is so nice and pretty, I think I am crying a little bit.

File: dependencies.yaml (hunk 1 above)
Comment (golovasteek) [reply dependencies.yaml:79]: Waiting to close the related PR, to get the right revision

Comment (golovasteek) [reply dependencies.yaml]: Updated

**Review by dorezyuk — APPROVED** [review]

**Pull request merged**